hbase-common/src/saveVersion.sh

```diff
@@ -30,34 +30,34 @@ nativeOutputDirectory="$2/native/utils/"
 pushd .
 cd ..
 
-user=`whoami | sed -n -e 's/\\\/\\\\\\\\/p'`
+user=$(whoami | sed -n -e "s/\\\/\\\\\\\\/p")
 if [ "$user" == "" ]
 then
-  user=`whoami`
+  user=$(whoami)
 fi
-date=`date`
-cwd=`pwd`
+date=$(date)
+cwd=$(pwd)
 if [ -d .svn ]; then
-  revision=`(svn info | sed -n -e 's/Last Changed Rev: \(.*\)/\1/p') || true`
-  url=`(svn info | sed -n -e 's/^URL: \(.*\)/\1/p') || true`
+  revision=$( (svn info | sed -n -e "s/Last Changed Rev: \(.*\)/\1/p") || true)
+  url=$( (svn info | sed -n -e 's/^URL: \(.*\)/\1/p') || true)
 elif [ -d .git ]; then
-  revision=`git log -1 --no-show-signature --pretty=format:"%H" || true`
-  hostname=`hostname`
+  revision=$(git log -1 --no-show-signature --pretty=format:"%H" || true)
+  hostname=$(hostname)
   url="git://${hostname}${cwd}"
 fi
 if [ -z "${revision}" ]; then
   echo "[WARN] revision info is empty! either we're not in VCS or VCS commands failed." >&2
   revision="Unknown"
   url="file://$cwd"
 fi
-if ! [  -x "$(command -v openssl)" ]; then
+if ! [ -x "$(command -v openssl)" ]; then
   if ! [ -x "$(command -v gpg)" ]; then
     srcChecksum="Unknown"
   else
-    srcChecksum=`find hbase-*/src/main/ | grep -e "\.java" -e "\.proto" | LC_ALL=C sort | xargs gpg --print-md sha512 | gpg --print-md sha512 | cut -d ' ' -f 1`
```

Comment:
Previous use of `cut` here at the end is surprising. It took only the first block...

Reply:
Based on history it was used for `md5` and `md5sum` commands but it wasn't removed when SHA512 was introduced in saveVersion.sh.

```
root@8224f4c69e6f:/# echo "foo" | md5sum
d3b07384d113edec49eaa6238ad5ff00  -
```

Reply:
Does `tr -d '[:space:]'` work? It's simpler and shorter than `tr '\n' ' ' | sed 's/[[:space:]]*//g'`

Reply:
That's wired. I did not see in my macOS and Ubuntu Linux. But since this is not reliable, I guess we can stay with your script `tr` and `sed` - which works for me on both macOS and Ubuntu.